src/Microsoft.AspNet.Http.Extensions/UriHelper.cs

```diff
@@ -25,7 +25,7 @@ public static class UriHelper
             FragmentString fragment = new FragmentString())
         {
             string combinePath = (pathBase.HasValue || path.HasValue) ? (pathBase + path).ToString() : "/";
-            return combinePath + query + fragment;
+            return $"{combinePath}{query.ToString()}{fragment.ToString()}";
```

Comment:
Since this has just 3 components, it would be more efficient reverting to the `String.Concat()` a+b+c format it was in before. While maintaining equal readability, it's ~3-4x faster on most machines (I've tested this on a variety locally and on the latest servers).